static/app/components/alerts/snoozeAlert.tsx

```diff
@@ -0,0 +1,115 @@
+import styled from '@emotion/styled';
+import * as Sentry from '@sentry/react';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import {IconChevron, IconMute, IconSound} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type Props = {
+  isSnoozed: boolean;
+  onSnooze: (nextState: {
+    snooze: boolean;
+    snoozeCreatedBy?: string;
+    snoozeForEveryone?: boolean;
+  }) => void;
+  projectSlug: string;
+  ruleId: string;
+};
+
+function SnoozeAlert({isSnoozed, onSnooze, projectSlug, ruleId}: Props) {
+  const organization = useOrganization();
+  const api = useApi();
+  function handleMute(target: string) {
+    try {
+      api.requestPromise(
+        `/projects/${organization.slug}/${projectSlug}/rules/${ruleId}/snooze/`,
+        {
+          method: 'POST',
+          data: {
+            target,
+          },
+        }
+      );
+    } catch (err) {
+      Sentry.captureException(err);
```

Comment:
Because the `requestPromise` is not awaited, this line will never be run. The promise will get created and it will immediately go to the `onSnooze` line

Reply:
In addition, you should handle errors. Think about what you want your user experience to be. You could either:

1. Disable the mute button while the request is running, then move to the success state once it has successfully finished.
2. Optimistically update the button to the success state. This means that in the event of an error, you want to revert to the previous state and add a toast with `addErrorMessage()`